toolchain/check/check_unit.cpp

```diff
@@ -346,6 +346,11 @@ auto CheckUnit::ProcessNodeIds() -> bool {
     node_id = *maybe_node_id;
     auto parse_kind = context_.parse_tree().node_kind(node_id);
 
+    if (context_.parse_tree().node_has_error(node_id)) {
+      context_.TODO(node_id, "handle invalid parse trees in `check`");
```

Comment:
Where this causes code to no longer be reached, would it be worth switching to `CARBON_FATAL`?

e.g.:

- HandleInvalidParse (and variants)
- The pre-existing TODOs for "Error recovery from keyword name"

(for contrast, valid code encounters "function with positional parameters", you're only removing 1 case of 4)

Reply:
Ah, yep - done.
I did change the "Error recovery from keyword name" to `CARBON_CHECK` since otherwise they'd be branch-to-unreachable (which is disfavorable in the LLVM style, at least) - happy to switch back to `CARBON_FATAL` if it makes more sense in this particular case.

Reply:
Makes sense to CHECK where it fits.